compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -268,7 +273,7 @@ class Semantic {
   }
 
   /** The state that threads through the interpreter */
-  type Contextual[T] = (Env, Context, Trace, Promoted) ?=> T
+  type Contextual[T] = (Env, (evalCache, evalCache), Context, Trace, Promoted) ?=> T
```

Comment:
In Scala, types need to be capitalized. Meanwhile, opaque types can help here to make the interface nicer.